test/units/TEST-03-JOBS.sh

```diff
@@ -166,4 +168,11 @@ assert_rc 3 systemctl --quiet is-active succeeds-on-restart.target
 systemctl start fails-on-restart.target || :
 assert_rc 3 systemctl --quiet is-active fails-on-restart.target
 
+# Test shortcutting auto restart
+
+(! systemctl start shortcut-restart.service)
+timeout 15 bash -c 'while [[ "$(systemctl show shortcut-restart.service -P SubState)" != "auto-restart" ]]; do sleep .5; done'
+systemctl start --no-block shortcut-restart.service
+timeout 15 bash -c 'while [[ "$(systemctl show shortcut-restart.service -P SubState)" != "start-pre" ]]; do sleep .5; done'
```

Comment:
Shouldn't we just check here that we leave the `auto-restart` state?

Reply:
> Shouldn't we just check here that we leave the auto-restart state?

We need to actually check that start job did run, rather than stuck in `auto-restart-queud` or job simply fails.